pennylane/transforms/qcut.py

```diff
@@ -210,3 +212,80 @@ def tape_to_graph(tape: QuantumTape) -> MultiDiGraph:
             order += 1
 
     return graph
+
+
+def fragment_graph(graph: MultiDiGraph) -> Tuple[Tuple[MultiDiGraph], MultiDiGraph]:
+    """
+    Fragments a graph into a collection of subgraphs as well as returning
+    the communication/[quotient](https://en.wikipedia.org/wiki/Quotient_graph) graph.
+
+    Args:
+        graph (MultiDiGraph): directed multigraph containing measure and prepare
+            nodes at cut locations
+
+    Returns:
+        subgraphs, communication_graph (Tuple[Tuple[MultiDiGraph], MultiDiGraph]):
+        the subgraphs of the cut graph and the communication graph where each
+        node represents a fragment and edges denote the flow of qubits between
+        fragments
+
+    **Example**
+
+    Consider the following circuit with manually-placed wire cuts:
+
+    .. code-block:: python
+
+        from pennylane.transforms import qcut
+
+        wire_cut_0 = qml.WireCut(wires=0)
+        wire_cut_1 = qml.WireCut(wires=1)
+        multi_wire_cut = qml.WireCut(wires=[0, 1])
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(0.4, wires=0)
+            qml.apply(wire_cut_0)
+            qml.RY(0.5, wires=0)
+            qml.apply(wire_cut_1)
+            qml.CNOT(wires=[0, 1])
+            qml.apply(multi_wire_cut)
+            qml.RZ(0.6, wires=1)
+            qml.expval(qml.PauliZ(0))
+
+    We can find the corresponding graph, remove all the wire cut nodes, and
+    find the subgraphs and communication graph by using:
+
+    >>> graph = qcut.tape_to_graph(tape)
+    >>> qcut.replace_wire_cut_nodes(graph)
+    >>> qcut.fragment_graph(graph)
+    ((<networkx.classes.multidigraph.MultiDiGraph object at 0x7fb3b2311940>,
+      <networkx.classes.multidigraph.MultiDiGraph object at 0x7fb3b2311c10>,
+      <networkx.classes.multidigraph.MultiDiGraph object at 0x7fb3b23e2820>,
+      <networkx.classes.multidigraph.MultiDiGraph object at 0x7fb3b23e27f0>),
+     <networkx.classes.multidigraph.MultiDiGraph object at 0x7fb3b23e26a0>)
+    """
+
+    edges = list(graph.edges)
+    cut_edges = []
+
+    for node1, node2, _ in edges:
+        if isinstance(node1, MeasureNode):
+            assert isinstance(node2, PrepareNode)
+            cut_edges.append((node1, node2))
+            graph.remove_edge(node1, node2)
+
+    subgraph_nodes = weakly_connected_components(graph)
+    subgraphs = tuple(graph.subgraph(n) for n in subgraph_nodes)
+
+    communication_graph = MultiDiGraph()
+    communication_graph.add_nodes_from(range(len(subgraphs)))
```

Comment:
Not a real suggestion but just wanted to throw it out there. Instead of constructing a graph with integer nodes, if we were to stick the subgraphs into the nodes' `data` attributes for the `communication_graph` (or, if possible, directly use subgraphs as nodes), we would only need to return the `communication_graph` which carries the subgraphs inside it. But again I would imagine it to be quite difficult to use with nx's default UI.

Reply:
We also discussed this possibility, we came to the conclusion that there were fairly equal pros and cons so it came down to that fact that it's already coded up, would interfere with the current tests and we're pushing to implement quickly. 

Happy to be persuaded otherwise though :grin: